x/signal/keeper.go

```diff
@@ -63,13 +63,6 @@ func (k Keeper) SignalVersion(ctx context.Context, req *types.MsgSignalVersion)
 		return nil, err
 	}
 
-	// The signalled version must be either the current version or the next
-	// version.
-	currentVersion := sdkCtx.BlockHeader().Version.App
-	if req.Version != currentVersion && req.Version != currentVersion+1 {
-		return nil, types.ErrInvalidVersion
-	}
```

Comment:
I think we should still assert that:

```go
req.Version >= currentVersion
```

so as to not support downgrades

Reply:
Yea good point so that we match this constraint: https://github.com/rootulp/celestia-app/blob/c270c424020b146168a10361c190eae4b073dab0/app/app.go#L468-L472

Reply:
Fixed

Reply:
The reason for wanting to allow to vote for the current version is: 
- Network is on v2
- We release v3
- Nodes start signalling for v3
- Bug is realised or for whatever reason we decide the release is not ready
- _Nodes signal back to the current version to cancel the upgrade_

I get your point about the delay but I think that can be overcome i.e. we write it as a noop that also does not trigger the delay

Reply:
ahh makes sense. An alternative would be:

Introduce a new transaction type that allows a validator to clear their previous signal.
Pro: clearer than signaling for current version
Con: larger change than allowing validators to signal for current version

We can allow voting for the current version and do a no-op for the delay if quorum version == current version

Reply:
Yeah I would be fine with that, like a `MsgCancelSignal` which under the hood signals for the current version